buildSrc/subprojects/profiling/src/main/kotlin/org/gradle/gradlebuild/profiling/buildscan/BuildScanPlugin.kt

```diff
@@ -102,46 +103,58 @@ open class BuildScanPlugin : Plugin<Project> {
     private
     fun Project.extractVcsData() {
 
-        fun Project.run(vararg args: String): String {
-            val process = ProcessBuilder(args.toList())
+        fun system(vararg args: String): String =
+            ProcessBuilder(args.toList())
                 .directory(rootDir)
                 .start()
-            assert(process.waitFor() == 0)
-            return process.inputStream.bufferedReader().use { it.readText().trim() }
-        }
+                .run {
+                    assert(waitFor() == 0)
+                    inputStream.bufferedReader().use { it.readText().trim() }
+                }
 
-        fun execAsync(action: () -> Unit) {
-            val latch = CountDownLatch(1)
-            thread(start = true) {
+        val threadPool = Executors.newCachedThreadPool()
```

Comment:
do we need to explicitly shutdown the threadpool now so we don't leak thread pools across builds?